### ISSUE TITLE
DWF CVE ID coverage - software, hardware, services, what else?

### ISSUE THREAD

**Issue by kurtseifried** [distributedweaknessfiling/dwf-workflow#8]
Currently, DWF CVE ID coverage is for software, hardware, and services which is in line with legacy CVE coverage.

What else should be covered moving forwards?

I think documentation is a prime contender, especially "official" documentation and widely used/popular documentation (e.g. StackOverflow answers?).

Also legacy CVE doesn't cover issues in malware typically, however, 1) who defines what is malware (e.g. state-sponsored malware), 2) who cares what the "intent:" of the code is? Code that runs on systems and has weaknesses and exposures should be covered by the legacy CVE system and DWF CVE IDs. So I think we should allow coverage for any software no matter what the intent is.

Anything else?

**Comment by joshbressers** [comment]
I don't like trying to rigidly define something like this. I prefer a more "interpretive" stance. That way as the world changes, what something means can change too.

The purpose of a security identifier is to give the security industry a common language to discuss a topic.

Historically we've boxed ourselves into the topic being software vulnerabilities. Even our request form reflects this mindset.

**Comment by kurtseifried** [comment]
Agreed, but I think it makes sense to define the things we definitely DO cover at least, and then expand scope as needed. I also think if we refuse to cover something as "out of scope" we should explain why in the issue and add it to a list because 1) we may decide at a later date that that type of thing falls in scope and 2) to track what we're saying no to (we may be biased, there may be something new we should cover, etc.). 

I think using an "OUTOFSCOPE" tag for the data would be the simplest and make things easy to find.

**Comment by deveyus** [comment]
I agree that covering all software, regardless of intent is a good idea, covering official documentation is also an intelligent choice, however I suspect that the traceability of most non-official sources would lead to functionally unusable data.  If there is an issue filed about a stack overflow answer, how would anyone utilize this information?  You'll never get users to check stack overflow against DWF before using the code (though potentially a browser extension could do so?) and once someone reads/copies the code from StackOverflow or a blog, they typically don't include attribution, so you would have no way to know unless we had some form of BNF-based expression for finding them automatically (which would be _amazing_ and useful for many things.) they're likely never going to be used.

**Comment by joshbressers** [comment]
This is one of the big questions

One possible explanation is to treat something like an issue in Stack Overflow as a discoverability issue. Think of it like a parts catalog. Then an automated scanner could look for a certain copypasta. By giving it an ID, while not directly actionable, it would make it possible for a future tool to do something useful.

**Comment by kurtseifried** [comment]
Beyond discoverability: Automated code scanners can go to that StackOverflow entry and add it to their data and then warn a user when they use a vulnerable code/configuration snippet for example. Also ideally we'll extend the data format to include better data (e.g. the code snippet/config issue itself) so it can be consumed directly. Part of this is also research, we need a corpus of tagged data to help train these ML models.

**Comment by kurtseifried** [comment]
What tags do we want to see, and keep in mind an identifier can have more than one tag:

1. Software
2. Firmware
3. Hardware
4. OpenSource
5. Proprietary
6. Documentation
7. Configuration
8. CodeExample

and so on?

**Comment by deveyus** [comment]
I think the continuation of that would include also the class of vulnerability "Overflow" "Logic" "Input Validation" etc.  Being able to get some broad classes of information would help with being able to work through them.
